eng/SnippetGenerator/MarkdownProcessor.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace SnippetGenerator
+{
+    public class MarkdownProcessor
```

Comment:
DocFX already supported this: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#code-snippet

Seems we shouldn't invent our own system we have to maintain when we already use DocFX.

Reply:
I don't want to turn README.md into templates and have another version be fully generated my goal is to keep them up-to-date.

Github doesn't understand `[!code-<language>[<name>](<codepath><queryoption><queryoptionvalue> "<title>")]` in markdowns so to use it it would require us to turn README.md into templates and run generators every time anything is eddited, not just snippets.

Reply:
And we don't use docfx for readmes.

Reply:
We will eventually use individual package readmes as the landing page for the docs. So we will use readmes in DocFx.

Reply:
But they would need to also work in Github. and `[!code-<language>[<name>](<codepath><queryoption><queryoptionvalue> "<title>")]` turns into [!code-<language>[<name>](<codepath><queryoption><queryoptionvalue> "<title>")] in github flawored markdown.

Reply:
Discussed offline. Since this is only for READMEs, DocFX has no real bearing on this change.